dockerspawner/dockerspawner.py

```diff
@@ -186,33 +188,41 @@ def poll(self):
         if not container:
             self.log.warn("container not found")
             raise gen.Return(0)
-        status = container['Status']
-        self.log.debug("Container %s status: %s", self.container_id[:7], status)
-        parts = status.split()
-        # status examples:
-        # 'Exited (127) 12 days ago'
-        # 'Up 4 days'
-        # 'Up 4 days (Paused)' <- TODO: this isn't handled
-        # e.g.
-        if not parts or parts[0] not in ('Up', 'Exited'):
-            raise ValueError("Unhandled status: '%s'" % status)
-        if parts[0] == 'Up':
+
+        container_state = container['State']
+        self.log.debug(
+            "Container %s status: %s",
+            self.container_id[:7],
+            pformat(container_state),
+        )
+
+        if container_state["Running"]:
             raise gen.Return(None)
-        elif parts[0] == 'Exited':
-            raise gen.Return(int(parts[1][1:-1]))
+        else:
+            raise gen.Return(
```

Comment:
Since we require Python 3, we can just `return` instead of using `gen.Return`. There are some artifacts leftover from when I started the project, and was still considering support for Python 2.